libs/net/traffic_generator.py

```diff
@@ -115,5 +136,47 @@ def _is_process_running(  # type: ignore[return]
         return False
 
 
-def is_tcp_connection(server: Server, client: Client) -> bool:
+class PodClient(BaseClient):
+    """Represents a TCP client that connects to a server to test network performance.
+
+    Expects pod to have iperf3 container.
+
+    Args:
+        pod (Pod): The pod where the client runs.
+        server_ip (str): The destination IP address of the server the client connects to.
+        server_port (int): The port on which the server listens for connections.
+        bind_interface (str): The interface or IP address to bind the client to (optional).
+            If not specified, the client will use the default interface.
+    """
+
+    def __init__(self, pod: Pod, server_ip: str, server_port: int, bind_interface: str | None = None):
+        super().__init__(server_ip=server_ip, server_port=server_port)
+        self._pod = pod
+        self._container = _IPERF_BIN
+        self._cmd += f" --bind {bind_interface}" if bind_interface else ""
+
+    def __enter__(self) -> "PodClient":
+        # run the command in the background using nohup to ensure it keeps running after the exec session ends
+        self._pod.execute(
+            command=["sh", "-c", f"nohup {self._cmd} >/tmp/{_IPERF_BIN}.log 2>&1 &"], container=self._container
+        )
+        self._ensure_is_running()
+
+        return self
+
+    def __exit__(self, exc_type: BaseException, exc_value: BaseException, traceback: object) -> None:
+        self._pod.execute(
+            command=["pkill", "-f", self._cmd],
+        )
+
+    def is_running(self) -> bool:
+        out = self._pod.execute(command=["pgrep", "-f", self._cmd], ignore_rc=True)
+        return bool(out.strip())
+
+    @retry(wait_timeout=30, sleep=2, exceptions_dict={})
```

Comment:
1. if you want all exceptions, nothing should be passed
2. TimeoutSampler is setting it to `{Exception: []}` - did you test `exceptions_dict={}` ? 

```
self.exceptions_dict = exceptions_dict if exceptions_dict is not None else {Exception: []}
```

Reply:
It is correctly set now.
When an empty dict is set, it means that any exception will fail imidiatly and that is the desired thing here.
When nothing is set, it defaults to what you mentioned above and will mean that all exceptions are retried, which is not what we want.

Per the response [here](https://github.com/RedHatQE/openshift-virtualization-tests/pull/2121#discussion_r2445311343), it was tested.